hypervisor/include/arch/x86/guest/vioapic.h

```diff
@@ -37,15 +37,16 @@
 #define	VIOAPIC_SIZE	4096UL
 
 #define REDIR_ENTRIES_HW	120U /* SOS align with native ioapic */
+#define ASSERT_BITMAP_SIZE	INT_DIV_ROUNDUP(REDIR_ENTRIES_HW, 64U)
 
 struct acrn_vioapic {
 	struct vm	*vm;
 	spinlock_t	mtx;
 	uint32_t	id;
 	uint32_t	ioregsel;
 	union ioapic_rte rtbl[REDIR_ENTRIES_HW];
-	/* sum of pin asserts (+1) and deasserts (-1) */
-	int32_t acnt[REDIR_ENTRIES_HW];
+	/* assert status bitmap: 1 - assert, 0 - deassert */
+	uint64_t assert[ASSERT_BITMAP_SIZE];
```

Comment:
level is better than assert.  pin_level/pin_state is fine too.

Reply:
will update